lib/eventsource.js

```diff
@@ -277,6 +277,10 @@ function EventSource (url, eventSourceInitDict) {
     })
 
     if (req.setNoDelay) req.setNoDelay(true)
+
+    // https://tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html
+    if (req.setSocketKeepAlive) req.setSocketKeepAlive(true)
```

Comment:
This function actually accepts an [optional second argument](https://nodejs.org/api/http.html#requestsetsocketkeepaliveenable-initialdelay). We should probably add support for that.

Reply:
Thanks for reviewing this PR and your feedback! I added 1s as default initial delay value. Later when options get exposed, we can add more logic here.